contracts/rules/RoboCop.sol

```diff
@@ -247,13 +247,16 @@ contract RoboCop is IAssetIO, Ownable, Pausable, ReentrancyGuard {
         return keccak256(abi.encode(triggers, actions, msg.sender, block.timestamp));
     }
 
-    // WARNING: only the last trigger's data gets sent back as triggerData
-    function _checkTriggers(Trigger[] storage triggers) internal view returns (bool valid, uint256 triggerData) {
+    function _checkTriggers(Trigger[] storage triggers) internal view returns (bool, TriggerReturn[] memory) {
+        TriggerReturn[] memory triggerReturnArr = new TriggerReturn[](triggers.length);
+        TriggerReturn memory triggerReturn;
+        bool valid = false;
         for (uint256 i = 0; i < triggers.length; i++) {
-            (valid, triggerData) = ITrigger(triggers[i].callee).check(triggers[i]);
-            if (!valid) return (false, 0);
+            (valid, triggerReturn) = ITrigger(triggers[i].callee).check(triggers[i]);
+            triggerReturnArr[i] = triggerReturn;
+            if (!valid) return (false, triggerReturnArr);
```

Comment:
if you return here, it the returnArr size wont match the input size. that seems icky but it does save gas fees. need to add warning comments saying so.

Reply:
the returnArr size will match input size. It's just that the ones after `i` will not have any TriggerReturn inside it (which is fine because the TriggerType will never match anything)